### PR TITLE
fix: infracost report

### PR DESCRIPTION
**Pull request by lemeurherve**
- Tested here: https://github.com/jenkins-infra/aws/pull/133/
- Related to https://github.com/jenkins-infra/helpdesk/issues/2852

### CODE REVIEW

File: vars/terraform.groovy [hunk 1]
```diff
@@ -107,20 +107,21 @@ def call(userConfig = [:]) {
...
                   title: "Terraform plan for this change request",
                   text: "The terraform plan for this change request can be found at <${planFileUrl}>.",
                   detailsURL: planFileUrl
               }
               stage('💸 Report estimated costs') {
-                try {
+                withCredentials([string(credentialsId: 'infracost-api-key', variable: 'INFRACOST_API_KEY')]) {
                   Boolean commentReport = false
                   Boolean commentComparison = false
+                  sh 'env|sort' // debut CHANGE_URL empty
```
Comment (lemeurherve) [inline vars/terraform.groovy:120]:
```suggestion
```

**Review by smerle33 — APPROVED** [review]
🚀

File: vars/terraform.groovy [hunk 2]
```diff
@@ -107,20 +108,19 @@ def call(userConfig = [:]) {
...
-                final String planFileUrl = "${env.BUILD_URL}artifact/terraform-plan-for-humans.txt"
+                final String planFileUrl = "${env.BUILD_URL}artifact/${planFileName}"
                 publishChecks name: "terraform-plan",
                   title: "Terraform plan for this change request",
                   text: "The terraform plan for this change request can be found at <${planFileUrl}>.",
                   detailsURL: planFileUrl
               }
               stage('💸 Report estimated costs') {
-                try {
+                withCredentials([string(credentialsId: 'infracost-api-key', variable: 'INFRACOST_API_KEY')]) {
```
Comment (dduportal) [inline vars/terraform.groovy:118]: Could you work around adding back the `try{}` block again please?
Anything going wrong in the infracost process should not impact our ability to execute terraform steps

**Review by dduportal — APPROVED** [review]
Approved, under the assumption that another "test build" is executed on jenkins-infra/aws to validate the last syntax

**Pull request merged**